File: two_sum.java
// Time Complexity: O(n)
// Space Complexity O(n)
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map_index = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int compl = target - nums[i];
            if(map_index.containsKey(compl)) return new int[] {map_index.get(compl), i};
            map_index.put(nums[i], i);

        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

File: two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        number_map = {}
        for i, n in enumerate(nums):
            comp = target - n
            if comp in number_map:
                return [number_map[comp], i]
            number_map[n] = i
        return []
